File: Python/TREE/Count Non-Leaf Nodes in Tree.py
//https://www.geeksforgeeks.org/problems/count-non-leaf-nodes-in-tree/1
class Solution:
    def countNonLeafNodes(self, root):
        # add code here
        if root is None:
            return 0
        if root.left is None and root.right is None:
            return 0
        return 1 + self.countNonLeafNodes(root.left) + self.countNonLeafNodes(root.right)

File: Java/TREE/Count Non-Leaf Nodes in Tree.java
//https://www.geeksforgeeks.org/problems/count-non-leaf-nodes-in-tree/1
class Solution
{
	int countNonLeafNodes(Node root) {
	    //code here
	    if(root==null){
	        return 0;
	    }
	    if(root.left==null && root.right==null){
	        return 0;
	    }
	    return 1+countNonLeafNodes(root.left)+countNonLeafNodes(root.right);
	    
	}
}
